src/main/java/com/epam/grid/engine/provider/utils/slurm/healthcheck/ShowConfigCommandParser.java

```diff
@@ -0,0 +1,159 @@
+/*
+ *
+ *  * Copyright 2022 EPAM Systems, Inc. (https://www.epam.com/)
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ *
+ */
+
+package com.epam.grid.engine.provider.utils.slurm.healthcheck;
+
+import com.epam.grid.engine.entity.CommandResult;
+import com.epam.grid.engine.entity.healthcheck.GridEngineStatus;
+import com.epam.grid.engine.entity.healthcheck.HealthCheckInfo;
+import com.epam.grid.engine.entity.healthcheck.StatusInfo;
+import com.epam.grid.engine.exception.GridEngineException;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.http.HttpStatus;
+
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.epam.grid.engine.utils.TextConstants.SPACE;
+import static com.epam.grid.engine.utils.TextConstants.EMPTY_STRING;
+import static com.epam.grid.engine.utils.TextConstants.EQUAL_SIGN;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ShowConfigCommandParser {
+
+    private static final String UP_STATE = "UP";
+    private static final String DOWN_STATE = "DOWN";
+
+    private static final String CANT_FIND_CONNECTION = "can't find connection";
+    private static final String CANT_FIND_CONNECTION_MESSAGE = "Can`t find connection via specified port";
+    private static final String DATE_FORMAT_CHECK_TIME = "MM/dd/yyyy HH:mm:ss:";
+    private static final String DATE_FORMAT_START_TIME = "yyyy-MM-dd'T'HH:mm:ss";
+    private static final String BOOT_TIME = "BOOT_TIME";
+    private static final Long NOT_PROVIDED = 99999L;
+    private static final String DOUBLED_SPACES_REGEX = "\\s+";
+
+    public static HealthCheckInfo parseShowConfigResult(final CommandResult commandResult) {
+        validateShowConfigResponse(commandResult);
```

Comment:
what will happen if exit code != 0 but it's not related to connection?

Reply:
method was changed a little bit